refinery/core/models.py

```diff
@@ -1434,35 +1337,22 @@ def galaxy_cleanup(self):
                     logger.error(error_msg, 'library', self.name, e.message)
 
             # Delete Galaxy Workflow
-            # NOTE: Legacy analysis launches depended on uploading a copy of
-            # the Workflow into galaxy, while newer Tool-based ones utilize the
-            # original galaxy workflow which we don't want to delete
-            if not self.is_tool_based:
-                if self.workflow_galaxy_id:
+            workflow_tool = tool_manager.models.WorkflowTool
+            try:
+                tool = workflow_tool.objects.get(analysis__uuid=self.uuid)
+            except(workflow_tool.DoesNotExist,
+                   workflow_tool.MultipleObjectsReturned) as e:
+                logger.error("Could not properly fetch Tool: %s", e)
```

Comment:
Is logging an error sufficient?

Reply:
@mccalluc Now that I look at the surrounding code the exceptions would be pretty odd to hit, but I don't think we should re-raise anything and stop histories from being deleted as well.